test/index.js

```diff
@@ -41,796 +36,679 @@ const expect = lab.expect;
 
 describe('Registration', () => {
 
-    it('should register', (done) => {
+    it('should register', async () => {
 
-        const server = new Hapi.Server();
-        server.connection({ host: 'test' });
+        const server = Hapi.server();
 
-        internals.bootstrapServer(server, require('../'), () => {
+        await internals.bootstrapServer(server, require('../'));
 
-            const routes = server.table();
-            expect(routes).to.have.length(1);
-            expect(routes[0].table).to.have.length(2);
-            done();
-        });
+        const routes = server.table();
+        expect(routes).to.have.length(2);
     });
 
-    it('should register with options', (done) => {
+    it('should register with options', async () => {
 
-        const server = new Hapi.Server();
-        server.connection({ host: 'test' });
-
-        internals.bootstrapServer(server, {
-            register: require('../'),
-            options: {
-                helpersPath: Path.join(__dirname, '../templates/helpers'),
-                cssPath: null,
-                endpoint: '/'
-            }
-        }, (err) => {
+        const server = Hapi.server();
 
+        try {
```

Comment:
You don't really need try/catch with that kind of expect, if it throws it will fail the test so just not having it is equivalent.

Reply:
What do you suggest instead, put await in a `const res` and just put that in the expect?

Reply:
I coded it that way because it made the purpose of the test more obvious.

Reply:
@Marsup should we leverage a pattern like [this](https://github.com/hapijs/code/blob/master/API.md#await-rejecttype-message)?

Reply:
I thought it was a bit much to ask and was willing to do it after this PR, but if you want to tackle that I won't hold you.